Update episode 5 and 6

ALuesink opened the pull request:
Still in progress

ALuesink committed:
Update episode 6

ALuesink committed:
Update debug.md

ALuesink committed:
Update ep6

ALuesink committed:
Update episode 6

ALuesink committed:
Update ep6

ALuesink committed:
Migrated episode 5 - Documentation and Citation

ALuesink commented:
Episode 5 - Documentation and Citation is migrated to the intermediate tutorial (https://github.com/common-workflow-lab/cwl-intermediate-tutorial/pull/7)

ALuesink committed:
Add More information page

ALuesink commented:
I've added a new episode at the end called More information with links to useful websites

mr-c commented:
@gcapes Thanks for all your assistance! Can you review this PR? Since it is bigger I want to finish it before reviewing yours

gcapes commented:
Sure - I'll start on it tomorrow. FWIW my PRs focus on the earlier episodes so could be reviewed independently of this one.

gcapes commented on _episodes/debug.md:
```diff
@@ -10,16 +10,64 @@ objectives:
+If you encounter an error, the best practice is to run the workflow with the `--debug` flag.
+This will provide you with extensive information on the error you encounter.
+~~~
+cwltool --debug` CWL_SCRIPT.cwl
```
extraneous backtick after `--debug`

gcapes commented on _episodes/debug.md:
```diff
@@ -10,16 +10,64 @@ objectives:
+This error message will give you the type of error as well as the line of code that contains the error.
+Some of these errors will be explained in this episode. 
+
+A first step to check if your CWL script contains any errors, you can run the workflow with the `--validate` flag.
```
Improved wording suggestion: As a first step ...

gcapes commented on _episodes/debug.md:
```diff
@@ -10,16 +10,64 @@ objectives:
+![]({{page.root}}/fig/YAML_error_typo_variable.png)
+
+### Wiring error
+Wiring errors often occur when you forget to add an output from a workflows step to the `outputs` section.
```
workflows -> workflow's

gcapes commented on _episodes/debug.md:
```diff
@@ -10,16 +10,64 @@ objectives:
+
+### Format error
+Some files need a specific format that needs to be specified in the YAML inputs file, for example the fastq file in the RNA-seq analysis.
+When you don't specify a format, an error will occur. You can for example use the [EDAM](https://www.ebi.ac.uk/ols/ontologies/edam)ontology) ontology.
```
fix link formatting - should be:
`[EDAM](https://www.ebi.ac.uk/ols/ontologies/edam) ontology.`

gcapes reviewed (dismissed):
This looks like a good start!
I've made a few specific wording and typo comments, but in addition would it be useful to have a section on array vs single-item mismatch? That was in the original outline.

I would also suggest runnable examples here. Perhaps as exercises, and/or the input files for the error outputs shown.

Finally, I think that code output boxes would be preferable to screenshots. Certainly for maintainability - if someone else wants to add/modify the example it is likely their screenshots will look different.

gcapes committed:
Fix typos

gcapes committed:
Replace screenshots with code block examples

- Also show the input code which leads to the error
  so learners can reproduce them.

gcapes commented:
I've now fixed all the suggestions, except I can't reproduce the error from "Using tabs instead of spaces".

I also seem unable to push to this PR's branch, so I'll make a new PR to replace this one.

mr-c commented:
> I've now fixed all the suggestions, except I can't reproduce the error from "Using tabs instead of spaces".
> 
> I also seem unable to push to this PR's branch, so I'll make a new PR to replace this one.

Did you try `gh pr checkout 56` using https://cli.github.com/ ?

gcapes commented:
No, I've not used the github cli, but that command would only check out the branch (wouldn't it?), which I've already done. I've locally rebased onto gh-pages, and made the fixes.

mr-c commented:
> No, I've not used the github cli, but that command would only check out the branch (wouldn't it?), which I've already done. I've locally rebased onto gh-pages, and made the fixes.

It sets up the remote to push as well. Anyhow, next time

gcapes commented:
Look like I just figured it out :) I'd renamed my local branch, and it looks like that has to match the remote branch name.

gcapes commented:
Can someone suggest how to reproduce the tabs instead of spaces error?

mr-c commented:
@ALuesink do you still have the `error.cwl` file?

ALuesink commented:
> @ALuesink do you still have the `error.cwl` file?

Yes, I have it. However, I now see it has no tabs so it would not create the error. I've tried a file with tabs instead of spaces, but I don't get the `found character \t` error. This time I get `'NoneType' object has no attribute ...`. 
Is it possible the error has changed in the meantime?

gcapes commented:
Thanks for checking. I get the same error I when trying to reproduce an example with tabs. I'll add an example to this episode next week.


Thanks
Gerard
--
Gerard Capes
Research Applications, IT Services, University Of Manchester
Hours: Tues - Fri, 08:00 - 16:00

________________________________
From: ALuesink ***@***.***>
Sent: 21 January 2022 16:47
To: carpentries-incubator/cwl-novice-tutorial ***@***.***>
Cc: Gerard Capes ***@***.***>; Mention ***@***.***>
Subject: Re: [carpentries-incubator/cwl-novice-tutorial] Update episode 5 and 6 (PR #56)


@ALuesink<https://github.com/ALuesink> do you still have the error.cwl file?

Yes, I have it. However, I now see it has no tabs so it would not create the error. I've tried a file with tabs instead of spaces, but I don't get the found character \t error. This time I get 'NoneType' object has no attribute ....
Is it possible the error has changed in the meantime?

—
Reply to this email directly, view it on GitHub<https://github.com/carpentries-incubator/cwl-novice-tutorial/pull/56#issuecomment-1018680133>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AC66UPXFY2E73J7X6VLR253UXGE2VANCNFSM5JHVK57A>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

mr-c commented on _episodes/debug.md:
```diff
@@ -10,16 +10,389 @@ objectives:
+- Using tabs instead of spaces. In YAML files indentations are made using spaces, not tabs.
+  Errors caused by tabs will show `'NoneType' object has no attribute 'name'`.
+
+  ~~~
+  cwlVersion: v1.2
+  class: Workflow
+
```
Maybe a shorter example? This is currently longer than a screenful with not much advantage for that

gcapes replied:
There's a tradeoff between using the example from the rest of the lesson that learners will be familiar with vs using a 'hello world' example which would be shorter but unrelated. What do you think?

gcapes commented:
This is now ready for review/merge. The 'wiring' error I haven't done, because I can't actually run the workflow in this lesson. It crashes my laptop part way through.

I would think these would work well as exercises rather than examples, but have followed the pattern in the original submission for now.

mr-c committed:
Merge remote-tracking branch 'origin/gh-pages' into Episode5-6

mr-c committed:
Merge branch 'gh-pages' into Episode5-6
